README.rst

```diff
@@ -22,14 +32,117 @@ float32     Signed    32    8      -3.402823466385288598117041834e+38  3.4028234
+appear lexicographically after positive ones (the first bit being repsonsible for the sign and equals to one if the
+number is negative), and floating-point encoded values are sorted decreasingly for negative input.
+
+Upon storage, the order of the bytes array for both floating-point and integer numbers, depends on the machine's
+`endianness <https://en.wikipedia.org/wiki/Endianness>`_:
+  * on Little Endian machines, the Most Significant Byte (MSB) is the last of the array;
+  * on Big Endian machines, the MSB is the first.
+
+Our encoding scheme works as follows:
+  * integers are first encoded in a byte array following the machine's endianness and then, if the machine is Little
+    Endian, the array is mirrored; then, the first byte (which will now be the MSB irregardless of the machine's
+    endianness) is XOR'd with 10000000, hence the sign bit is flipped.
+    For example, for 8-bit integers, the result of two's complement is
+    [10000000 (-128), 10000001 (-127), ..., 11111111 (-1), 00000000 (0), 00000001 (1), ..., 01111111 (127)].
+
+    Applying our encoding scheme will result in the ordering
+    [00000000 (-128), 00000001 (-127), ..., 01111111 (-1), 10000000 (0), 10000001 (1), ..., 11111111 (127)].
+
+  * floating-point numbers are encoded in a byte array following the machine's endianness and then, if the machine is
+    Little Endian, the array is mirrored; then, if the number is nonnegative, the first byte (which will now be the MSB
+    irregardless of the machine's endianness) is OR'd with 0x80 (binary 10000000), hence the sign bit is set to 1
+    (treating the two signed zeroes as one); otherwise, the whole byte array is negated, which results in flipping
+    all bits and thus reverse-ordering negative numbers. Plus and minus infinity can be encoded in this scheme and will
+    come after and before all other byte arrays, respectively.
+
+Usage
+=====
+
+As a Pyhton library
+-------------------
+
+To use the encoding and decoding functions, you need to invoke one of the ``numenc.from_TYPE()`` and
+``numenc.to_TYPE()`` functions, replacing TYPE with any one of the types listed above. For example:
+
+.. code-block:: python
+
+    >>> import numenc
+
+    >>> numenc.from_int32(1200)
+    b'\x80\x00\x04\xb0'
+    >>> numenc.to_int32(b'\x80\x00\x04\xb0')
+    1200
+    >>> numenc.from_int32("a string")
+    Traceback (most recent call last):
+     ...
+    TypeError: Wrong input: expected signed 32-bit integer.
+
+    # negative input throws ValueError for unsigned types
+    >>> numenc.from_uint8(-1)
+    Traceback (most recent call last):
+     ...
+    ValueError: expected 8-bit unsigned integer (range [0, 255]), got -1.
+
+    >>> numenc.from_int64(-999999)
+    b'\x7f\xff\xff\xff\xff\xf0\xbd\xc1'
+    >>> numenc.to_int64(b'\x7f\xff\xff\xff\xff\xf0\xbd\xc1')
+    -999999
+
+    # small rounding differences are possible
```

Comment:
could you please add a comment explaining why this is so? because python internally uses larger floats (64-bits?)?

Reply:
indeed!